architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+
+**Mitigation:**
+
+1. **mTLS (Mutual TLS)**: Implement mutual TLS to ensure that both the client (Kubernetes API Server) and the server (Validating Webhook) authenticate each other. This helps in verifying the authenticity of both parties and prevents unauthorized access. Note that mTLS is enabled for the communication between the Validating Webhook and the Kubernetes API Server in Radius as of now.
+2. **Network Policies**: Use Kubernetes Network Policies to restrict the network traffic between the Kubernetes API Server and the webhook. This limits the exposure of the webhook to only trusted sources. Reference: <https://kubernetes.io/docs/concepts/services-networking/network-policies/>.
+
+#### An attacker gaining access to etcd and manually changing resources
+
+Risk: If an attacker gains access to etcd, they can manually change resources stored in the etcd datastore. This could lead to unauthorized access, data breaches, and manipulation of cluster operations.
+
+Mitigation:
+
+1. **Access Controls for etcd**: Implement strict access controls for etcd to ensure that only authorized components can access and modify its data.
+2. **Network and Firewall Policies**: Add the necessary Network and Firewall Policies to filter access to the etcd.
+3. **Audit Logs**
+4. **Encryption**
+
+#### Dangerous data hidden in the Recipe or Deployment resource
+
+Risk: When an external interactor sends a create or update request for a Recipe or Deployment, there is a risk that the interactor could hide dangerous data within the resource. This hidden data could potentially exploit vulnerabilities in the system, leading to security breaches or system malfunctions.
+
+Mitigation:
+
+1. **Use of Kubernetes Parser**: Radius relies on the parser provided by Kubernetes, which is well-tested and maintained. This reduces the risk associated with parsing malicious data.
+2. **Avoid Custom Parsers**: By not using custom or third-party parsers, Radius avoids the additional risk and complexity associated with ensuring the security of these parsers.
+3. **Fuzz Testing**: If a custom parser were to be used, it would be essential to implement fuzz testing. Fuzz testing involves providing invalid, unexpected, or random data inputs to the parser to identify potential vulnerabilities. For more information on fuzz testing, refer to [OWASP Fuzzing](https://owasp.org/www-community/Fuzzing).
+
+#### Users with access to the webhook server modifying the behavior of the webhook server
+
+Risk: Could an insider with access to the webhook server modify its behavior to approve malicious requests?
+
+Mitigation: Implement strict access controls, audit logs, and monitoring for changes to the webhook server configuration and code.
```

Comment:
This is a great one to discuss. It falls in the category of "what is a valid use case" and what is not.